res/controllers/Traktor-Kontrol-S2-MK1-hid-scripts.js

```diff
@@ -0,0 +1,1361 @@
+
+    MessageShort.addControl("[Channel2]", "!top_encoder_press", 0x0E, "B", 0x10, false, this.topEncoderPress);
+    MessageShort.addControl("[Channel2]", "!shift", 0x0C, "B", 0x80, false, this.shift);
+    MessageShort.addControl("[Channel2]", "!sync_enabled", 0x0C, "B", 0x40, false, this.syncButton);
+    MessageShort.addControl("[Channel2]", "!cue_default", 0x0C, "B", 0x20, false, this.cueButton);
+    MessageShort.addControl("[Channel2]", "!play", 0x0C, "B", 0x10, false, this.playButton);
+    MessageShort.addControl("[Channel2]", "!pad1", 0x0C, "B", 0x08, false, this.padButton);
+    MessageShort.addControl("[Channel2]", "!pad2", 0x0C, "B", 0x04, false, this.padButton);
+    MessageShort.addControl("[Channel2]", "!pad3", 0x0C, "B", 0x02, false, this.padButton);
+    MessageShort.addControl("[Channel2]", "!pad4", 0x0C, "B", 0x01, false, this.padButton);
+    MessageShort.addControl("[Channel2]", "!loop_in", 0x0B, "B", 0x40, false, this.loopInButton);
+    MessageShort.addControl("[Channel2]", "!loop_out", 0x0B, "B", 0x20, false, this.loopOutButton);
+    MessageShort.addControl("[Channel2]", "!remix_button", 0x0B, "B", 0x01, false, this.samplerModeButton);
+    MessageShort.addControl("[Channel2]", "!flux_button", 0x0B, "B", 0x10, false, this.introOutroModeButton);
+    MessageShort.addControl("[Channel2]", "!left_encoder_press", 0x0E, "B", 0x20, false, this.leftEncoderPress);
+    MessageShort.addControl("[Channel2]", "!right_encoder_press", 0x0E, "B", 0x40, false, this.rightEncoderPress);
+    MessageShort.addControl("[Channel2]", "!jog_wheel", 0x05, "I", 0xFFFFFFFF, false, this.jogMove);
+    MessageShort.addControl("[Channel2]", "!load_track", 0x0B, "B", 0x04, false, this.loadTrackButton);
+    MessageShort.addControl("[EffectRack1_EffectUnit2]", "!effect_focus_button",
+        0x0A, "B", 0x80, false, this.effectFocusButton);
+    MessageShort.addControl("[EffectRack1_EffectUnit2]", "!effectbutton1", 0xA, "B", 0x40, false, this.effectButton);
+    MessageShort.addControl("[EffectRack1_EffectUnit2]", "!effectbutton2", 0xA, "B", 0x20, false, this.effectButton);
+    MessageShort.addControl("[EffectRack1_EffectUnit2]", "!effectbutton3", 0xA, "B", 0x10, false, this.effectButton);
+
+    MessageShort.addControl("[Channel1]", "!pfl", 0x09, "B", 0x80, false, this.pflButton);
+    MessageShort.addControl("[EffectRack1_EffectUnit1]", "group_[Channel1]_enable", 0x0A, "B", 0x02);
+    MessageShort.addControl("[EffectRack1_EffectUnit2]", "group_[Channel1]_enable", 0x0A, "B", 0x01);
+
+    MessageShort.addControl("[Channel2]", "!pfl", 0x0B, "B", 0x80, false, this.pflButton);
+    MessageShort.addControl("[EffectRack1_EffectUnit1]", "group_[Channel2]_enable", 0x0A, "B", 0x08);
+    MessageShort.addControl("[EffectRack1_EffectUnit2]", "group_[Channel2]_enable", 0x0A, "B", 0x04);
+
+    MessageShort.addControl("[Master]", "maximize_library", 0x0E, "B", 0x08, false, this.toggleButton);
```

Comment:
The browse encoder press

Reply:
Please add a comment in the code, this is not self-explaining

Reply:
You might also try
```
if (shift) {
                engine.setValue("[Library]", "sort_column_toggle", 0);
            } else {
                engine.setValue("[Library]", "MoveFocusForward", 1);
            }
```
instead of maximize_library. This would allow to navigate through the library, and as shifted operation allows to sort by selected library column.

Reply:
`"MoveFocusForward"` does not work for me. But also pressing `TAB` does only highlight different sections in the library. How is this more useful?

Reply:
I tried it but it does not work. It just adds a marker around the left or the right window. Nothing more. Should you be able to navigate the left window?

Reply:
Yes, this should allow you to browse to the library tree on the left, to select another crate etc.
But I guess, this works only in 2.4.

Reply:
I use `"[Playlist]", "SelectTrackKnob"`. Maybe I should be using something like `[Library]` `MoveVertical`

Reply:
Try something like
```
if (shift) {
            engine.setValue("[Library]", "MoveHorizontal", delta);
        } else {
            engine.setValue("[Library]", "MoveVertical", delta);
        }
```

Reply:
I cannot figure out how to do what I want: Move to the next/prev song without shift and browse the sidebar with shift. There is no way to set a focus, just to set it to the next window :disappointed:.

Reply:
I guess this feature requires 2.4 work. Here this code snippets work with 2.4. Maybe you can provide a follow-up PR for 2.4 once this is merged.

Reply:
Yes `[Library]` `focus_widget` is what i am looking for.